### PR TITLE
Enhancement - Replace CLI code with Clap cargo

### PR DESCRIPTION
**Pull request by jaysc**
Relates to #17

Initial replacement of our CLI with [clap](https://docs.rs/clap/latest/clap/index.html).

### CODE REVIEW

**Comment by JonShort** [comment]
Pulled this down - seems to work well - I kinda expected it to remove a lot more code our side though, it's almost comparable

Does make future changes easier though so I'm open to the change

**Comment by jaysc** [comment]
I accidently included `cmd.rs` which was my replace file as I was working through out. It's been removed now.

I think it's great that it just handles all the arg parsing. We don't have to deal with the split and check the tag anymore. Also the help autogen with doc comments is nicely formatted.

There may be more we could do.

**Comment by JonShort** [comment]
Seems decent, think we should add - make it ready for review when happy with it

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,9 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-serde_json = "1"
+serde_json = "1.0.91"
```
Comment (JonShort) [inline Cargo.toml]: Prefer to leave as 1 really

```suggestion
serde_json = "1"
```

File: src/main.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use std::{env, process};
+use std::process::{self};
```
Comment (JonShort) [inline src/main.rs]:
```suggestion
use std::process;
```

File: Cargo.toml [hunk 3]
```diff
@@ -7,7 +7,9 @@ edition = "2018"
...
+clap = { version = "4.0.32", features = ["derive"] }
```
Comment (JonShort) [inline Cargo.toml]: 4.1.1 is out - https://github.com/clap-rs/clap/blob/master/CHANGELOG.md#411---2023-01-14

File: src/utility.rs [hunk 4]
```diff
@@ -1,6 +1,32 @@
...
+pub struct Args {
+    ///Bump dependencies to latest possible version (includes major changes)
+    #[arg(short, long)]
+    pub latest: bool,
+
+    #[arg(short, long)]
+    ///Update to latest patch version only
+    pub patch: bool,
+
+    #[arg(short = 'o', long)]
```
Comment (JonShort) [inline src/utility.rs]: No short would be fine here tbh - it's a bit of an escape hatch

**Review by JonShort — APPROVED** [review]

**Pull request merged**